### PR TITLE
(#1235) replaced  MatcherAssert.assertThat with cactoos.matchers

### PR DESCRIPTION
**Pull request by mohamednizar**
#1235 ,replaced `MatcherAssert.assertThat` with `cactoos.matchers.Assertion`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1238](https://github.com/yegor256/cactoos/issues/1238) is now in scope, role is `REV`

**Comment by mohamednizar** [comment]
@0crat wait

**Comment by 0crat** [comment]
> @0crat wait ([here](https://github.com/yegor256/cactoos/issues/1238#issuecomment-554468758))

@mohamednizar Job `gh:yegor256/cactoos#1238` is not assigned, can't get performer

**Comment by 0crat** [comment]
This pull request [#1238](https://github.com/yegor256/cactoos/issues/1238) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C63314D6Z/4d83745a-9b49-4c06-8a38-4fa8b3115721); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 1]
```diff
@@ -40,79 +40,84 @@
...
-            Matchers.equalTo(Arrays.copyOfRange(bytes, 1, bytes.length))
-        );
+        new Assertion<>(
+        "Can't tail on Long stream",
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
+                size - 1
+                    )
+                ).asBytes(),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar fix the indentation and put the brackets properly.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 2]
```diff
@@ -40,79 +40,84 @@
...
+                Matchers.equalTo(Arrays.copyOfRange(bytes, 1, bytes.length))
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar use `InputHasContent` as I've been said above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 3]
```diff
@@ -40,79 +40,84 @@
...
     @Test
     public void tailsOnExactStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Can't tail on Exact Stream",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 4]
```diff
@@ -40,79 +40,84 @@
...
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 5]
```diff
@@ -40,79 +40,84 @@
...
     public void tailsOnExactStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Can't tail on Exact Stream",
+        new BytesOf(
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 6]
```diff
@@ -40,79 +40,84 @@
...
     @Test
     public void tailsOnExactStreamAndBuffer() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Can't tails on Exact Stream and Buffer",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 7]
```diff
@@ -40,79 +40,84 @@
...
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 8]
```diff
@@ -40,79 +40,84 @@
...
+        ).affirm();
     }
 
     @Test
     public void tailsOnShorterStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
+        new Assertion<>(
+        "Can't tails on Shorter Stream",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 9]
```diff
@@ -40,79 +40,84 @@
...
     @Test
     public void tailsOnStreamLongerThanBufferAndBytes() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Can't tails on Stream longer than buffer and bytes",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 10]
```diff
@@ -40,79 +40,84 @@
...
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 11]
```diff
@@ -40,79 +40,84 @@
...
                     size - 1,
                     size - 1
                 )
             ).asBytes(),
             Matchers.equalTo(
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 12]
```diff
@@ -40,79 +40,84 @@
...
-        MatcherAssert.assertThat(
+        new Assertion<>(
+        "Can't tails on Shorter Stream",
             new BytesOf(
                 new TailOf(
                     new InputOf(new BytesOf(bytes)),
                     size + 1
                 )
             ).asBytes(),
             Matchers.equalTo(bytes)
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 13]
```diff
@@ -40,79 +40,84 @@
...
     @Test
     public void tailsOnShorterStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
+        new Assertion<>(
+        "Can't tails on Shorter Stream",
             new BytesOf(
                 new TailOf(
                     new InputOf(new BytesOf(bytes)),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 14]
```diff
@@ -40,79 +40,84 @@
...
+        "Can't tails on Exact Stream and Buffer",
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
                     size,
-                    size
+                 size
                 )
             ).asBytes(),
             Matchers.equalTo(bytes)
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 15]
```diff
@@ -40,79 +40,84 @@
...
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Can't tail on Exact Stream",
+        new BytesOf(
+            new TailOf(
+                new InputOf(new BytesOf(bytes)),
                     size
                 )
             ).asBytes(),
             Matchers.equalTo(bytes)
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar same above.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 16]
```diff
@@ -40,79 +40,84 @@
...
                 )
             ).asBytes(),
             Matchers.equalTo(
                 Arrays.copyOfRange(bytes, 1, bytes.length)
             )
-        );
+        ).affirm();
     }
 
     @Test(expected = IllegalArgumentException.class)
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar use cactoos-matchers' `Throws` instead.

**Comment by fabriciofx** [comment]
@mohamednizar ping

**Comment by mohamednizar** [comment]
> @mohamednizar ping

Just got some time to go through the review, I'll do the changes and commit them.

**Comment by fabriciofx** [comment]
@mohamednizar please, fix the CI failures.

**Comment by mohamednizar** [comment]
@fabriciofx , can you suggest me a tool for code formatting? I'm using Intellij IDE - community version.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 17]
```diff
@@ -40,79 +40,84 @@
...
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
-                    size - 1
-                )
-            ).asBytes(),
-            Matchers.equalTo(Arrays.copyOfRange(bytes, 1, bytes.length))
-        );
+        new Assertion<>(
+        "Can't tail on Long stream",
+        new BytesOf(
```
[the remark this answers is not in the document]
Comment (mohamednizar) [reply src/test/java/org/cactoos/io/TailOfTest.java]: `InputHasContent` not working with current implementation. shall we keep the current change?

[the remark this answers is not in the document]
Comment (mohamednizar) [reply src/test/java/org/cactoos/io/TailOfTest.java]: `org.hamcrest.core.IsEqual` working in this point

**Comment by fabriciofx** [comment]
@mohamednizar I'm using IntelliJ IDEA too (Enterprise Version). If you know a tool format code using Yegor's style code, please let me know! 😄 

P.S. Maybe you can use [astyle](http://astyle.sourceforge.net/) or something else to format?

**Comment by fabriciofx** [comment]
@mohamednizar it seems fine to me. @paulodamaso can you merge it, please?

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 18]
```diff
@@ -40,79 +40,82 @@
...
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
-                    size - 1
-                )
-            ).asBytes(),
-            Matchers.equalTo(Arrays.copyOfRange(bytes, 1, bytes.length))
-        );
+        new Assertion<>(
+        "Must tail on Long stream",
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar Let's change this assertion failure message to "Must return last part of Long stream", making it
somewhat closer to `tail` description on linx manual.

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 19]
```diff
@@ -40,79 +40,82 @@
...
     @Test
     public void tailsOnExactStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Must tail on Exact Stream",
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar Please see above about assertion failure messages

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 20]
```diff
@@ -40,79 +40,82 @@
...
     @Test
     public void tailsOnExactStreamAndBuffer() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
-            new BytesOf(
-                new TailOf(
-                    new InputOf(new BytesOf(bytes)),
+        new Assertion<>(
+        "Must tails on Exact Stream and Buffer",
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar Please see above about assertion failure messages

File: src/test/java/org/cactoos/io/TailOfTest.java [hunk 21]
```diff
@@ -40,79 +40,82 @@
...
+        ).affirm();
     }
 
     @Test
     public void tailsOnShorterStream() throws Exception {
         final int size = 4;
         final byte[] bytes = this.generate(size);
-        MatcherAssert.assertThat(
+        new Assertion<>(
+        "Must tails on Shorter Stream",
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/io/TailOfTest.java]: @mohamednizar Please see above about assertion failure messages

**Comment by fabriciofx** [comment]
@mohamednizar ping

**Comment by paulodamaso** [comment]
@mohamednizar ping

**Comment by paulodamaso** [comment]
@0crat assign me

**Comment by paulodamaso** [comment]
@mohamednizar Please go on wit the implementation

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/cactoos/issues/1238#issuecomment-581595725))

@paulodamaso This pull request [#1238](https://github.com/yegor256/cactoos/issues/1238) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/cc5ea965-ef59-4b32-b444-70356979f1aa); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/cactoos/issues/1238#issuecomment-581595725))

@paulodamaso There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@377a78bde909, thread: PQ-C63314D6Z
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [comment, job, issue, repo, cause, _parent_sig, flow, login]
```

[0.51.0](https://github.com/zerocracy/farm): CID: [c227bd8d-a98f-484b-870f-e53913b83808](https://www.0crat.com/footprint/C63314D6Z/c227bd8d-a98f-484b-870f-e53913b83808), Type: "Order was canceled"

**Comment by paulodamaso** [comment]
@mohamednizar I'll close it due inactivity

**Comment by 0crat** [comment]
Job was finished in 17 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Pull request [#1238](https://github.com/yegor256/cactoos/issues/1238) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Pull request closed without merging**